bin/faucet/src/main.rs

```diff
@@ -466,6 +456,20 @@ mod test {
             });
         });
 
+        // Wait for faucet to be up
+        let addr = config.endpoint.to_socket().unwrap();
+        let start = Instant::now();
+        let timeout = Duration::from_secs(10);
+        loop {
+            match tokio::net::TcpStream::connect(addr).await {
+                Ok(_) => break,
+                Err(_) if start.elapsed() < timeout => {
+                    sleep(Duration::from_millis(200)).await;
+                },
+                Err(e) => panic!("faucet never became reachable: {e}"),
+            }
+        }
```

Comment:
I added this waiting loop because it seems like there was a race condition where the client was attempting to connect before the faucet was fully up (since the faucet gets started on a different thread)? It was causing the website test to fail ([CI run](https://github.com/0xMiden/miden-node/actions/runs/15398251754/job/43324326150)). I don't know how this didn't come up before, maybe just worse luck now. cc @SantiagoPittella @TomasArrachea